test/powershell/Language/Scripting/CommandErrorHandling.Tests.ps1

```diff
@@ -0,0 +1,470 @@
+
+    BeforeAll {
+        $pwsh = [PowerShell]::Create()
+        $pwsh.AddScript(@'
+
+    function ThrowTerminatingError {
+        [CmdletBinding()]
+        param()
+
+        $ex = [System.ArgumentException]::new('terminating-exception')
+        $er = [System.Management.Automation.ErrorRecord]::new($ex, 'ThrowTerminatingError:error', 'InvalidArgument', $null)
+        $PSCmdlet.ThrowTerminatingError($er)
+
+        Write-Verbose -Verbose "verbose-message"
+    }
+
+    function ErrorActionStop {
+        [CmdletBinding()]
+        param()
+
+        Get-Command NonExist -ErrorAction Stop
+        Write-Verbose -Verbose "verbose-message"
+    }
+
+    function ThrowException {
+        [CmdletBinding()]
+        param()
+
+        throw 'throw-exception'
+        Write-Verbose -Verbose "verbose-message"
+    }
+
+    function WriteErrorAPI {
+        [CmdletBinding()]
+        param()
+
+        $ex = [System.ArgumentException]::new('arg-exception')
+        $er = [System.Management.Automation.ErrorRecord]::new($ex, 'WriteErrorAPI:error', 'InvalidArgument', $null)
+        $PSCmdlet.WriteError($er)
+
+        Write-Verbose -Verbose "verbose-message"
+    }
+
+    function WriteErrorCmdlet {
+        [CmdletBinding()]
+        param()
+
+        Write-Error 'write-error-cmdlet'
+        Write-Verbose -Verbose "verbose-message"
+    }
+
+    function MethodInvocationThrowException {
+        [CmdletBinding()]
+        param()
+
+        ## This method call throws exception.
+        $iss = [initialsessionstate]::Create()
+        $iss.ImportPSModule($null)
+
+        Write-Verbose -Verbose "verbose-message"
+    }
+
+    function ExpressionThrowException {
+        [CmdletBinding()]
+        param()
+
+        1/0 ## throw exception.
+        Write-Verbose -Verbose "verbose-message"
+    }
+
+'@).Invoke()
+
+        function RunCommand {
+            param(
+                [string] $Command,
+                [ValidateSet('Continue', 'Ignore', 'SilentlyContinue', 'Stop')]
+                [string] $ErrorAction
+            )
+
+            $pwsh.Commands.Clear()
+            $pwsh.Streams.ClearStreams()
+            $pwsh.AddCommand($command).AddParameter('ErrorAction', $ErrorAction).Invoke()
+        }
+
+        function RunScript {
+            param([string] $Script)
+
+            $pwsh.Commands.Clear()
+            $pwsh.Streams.ClearStreams()
+            $pwsh.AddScript($Script).Invoke()
+        }
+
+        function GetLastError {
+            $pwsh.Commands.Clear()
+            $pwsh.AddCommand('Get-Error').Invoke()
+        }
+
+        function ClearDollarError {
+            $pwsh.Commands.Clear()
+            $pwsh.AddScript('$Error.Clear()').Invoke()
+        }
+    }
+
+    AfterAll {
+        $pwsh.Dispose()
+    }
+
+    Context 'Terminating error' {
+
+        It "'ThrowTerminatingError' stops execution with 'ActionPreference.<ErrorAction>' for error action" -TestCases @(
```

Comment:
Based on previous explanation I'd reword the header (and below too):
```suggestion
        It "'ThrowTerminatingError' always stops execution without being affected the explicit 'ActionPreference.<ErrorAction>' error action" -TestCases @(
```

Reply:
I have updated all test case names, to make them more readable.